datavec/datavec-python/src/main/java/org/datavec/python/NumpyArray.java

```diff
@@ -0,0 +1,179 @@
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.bytedeco.javacpp.FloatPointer;
+import org.bytedeco.javacpp.DoublePointer;
+import org.nd4j.linalg.dataset.DataSet;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.nativeblas.NativeOps;
+import org.nd4j.nativeblas.NativeOpsHolder;
+import org.nd4j.linalg.api.buffer.DataType;
+
+
+import java.util.ArrayList;
+
+
+public class NumpyArray {
+
+    private static NativeOps nativeOps = NativeOpsHolder.getInstance().getDeviceNativeOps();
+    private long address;
+    private long[] shape;
+    private long[] strides;
+    private DataType dtype = DataType.FLOAT;
+    private INDArray nd4jArray;
+
+    public NumpyArray(long address, long[] shape, long strides[], boolean copy){
+        this.address = address;
+        this.shape = shape;
+        this.strides = strides;
+        setND4JArray();
+        if (copy){
+            nd4jArray = nd4jArray.dup();
+            this.address = nd4jArray.data().address();
+
+        }
+    }
+    public NumpyArray(long address, long[] shape, long strides[]){
+        this(address, shape, strides, false);
+    }
+
+    public NumpyArray(long address, long[] shape, long strides[], DataType dtype){
+        this(address, shape, strides, dtype, false);
+    }
+
+    public NumpyArray(long address, long[] shape, long strides[], DataType dtype, boolean copy){
+        this.address = address;
+        this.shape = shape;
+        this.strides = strides;
+        this.dtype = dtype;
+        setND4JArray();
+        if (copy){
+            nd4jArray = nd4jArray.dup();
+            this.address = nd4jArray.data().address();
+        }
+    }
+
+    public long getAddress() {
+        return address;
+    }
+
+    public long[] getShape() {
+        return shape;
+    }
+
+    public long[] getStrides() {
+        return strides;
+    }
+
+    public DataType getDType() {
+        return dtype;
+    }
+
+    public JSONObject toJSON(){
+        JSONObject jsonObject = new JSONObject();
+        JSONArray data = new JSONArray();
+        long size = 1;
+        for (long d: nd4jArray.shape()){
+            size *= d;
+        }
+        if (dtype == DataType.SHORT || dtype == DataType.INT || dtype == DataType.LONG){
+            for (int i=0; i < size; i++){
+                data.add((long)nd4jArray.getDouble(i));
+            }
+        }
+        else if (dtype == DataType.FLOAT || dtype == DataType.DOUBLE){
+            for (int i=0; i<size; i++){
+                data.add(nd4jArray.getDouble(i));
+            }
+        }
+        JSONArray shape = new JSONArray();
+        for (long d: this.shape){
+            shape.add(d);
+        }
+        String dtypeStr = dtype.toString();
+        jsonObject.put("data", data);
+        jsonObject.put("shape", shape);
+        jsonObject.put("dtype", dtypeStr);
+        return jsonObject;
+    }
+
+
+    public NumpyArray(JSONObject json) throws Exception{
+        address = (Long)json.get("address");
```

Comment:
This looks inefficient, we should be able to call Python functions directly to get these values without serializing everything to JSON, but I'm not sure how much of a slowdown this is in practice.